File: programmers/42577_2.cpp
#include <string>
#include <vector>

using namespace std;

bool check_data(string basis, string phone)
{
	if (basis.length() > phone.length())
		return false;
	for (int i = 0; i < basis.length(); i++)
	{
		if (basis[i] != phone[i])
			return false;
	}
	return true;
}

bool solution(vector<string> phone_book) {
    bool answer = true;

	for (int i = 0; i < phone_book.size(); i++)
	{
		for (int j = 0; j < phone_book.size(); j++)
		{
			if (i == j)
				continue;
			if (check_data(phone_book[i], phone_book[j]))
			{
				answer = false;
				return answer;
			}
		}
	}
    return answer;
}

File: programmers/42577_2.py
# 전화번호 목록
# https://programmers.co.kr/learn/courses/30/lessons/42577
def solution(phone_book):
    answer = True
    for i in range(len(phone_book)):
        for j in range(len(phone_book)):
            if i == j :
                continue
            if phone_book[i].startswith(phone_book[j]):
                answer = False
                return answer
    return answer

'''
    find, startswith
    https://dpdpwl.tistory.com/119
'''